file_process/csv.py

```diff
@@ -5,67 +5,65 @@
 import pandas as pd
 from numpy import number, nan
 from pandas.errors import ParserError
-from werkzeug.datastructures import FileStorage
 
 from file_process.base import FileProcessorBase
-from file_process.exceptions import ModelFileValidationTargetsError, ModelFileValidationVariablesError, DelimiterError
+from file_process.exceptions import NotAllTargetsError, ModelFileValidationVariablesError, DelimiterError, \
+    NotSomeTargetsError
 
 
 class CSVFileProcessor(FileProcessorBase):
-    def read_file(self, file, **kwargs):
-        if isinstance(file, FileStorage):  # TODO try to get rid of it
-            file = file.read()
-            file = BytesIO(file)
-        read_rows_count = kwargs.get('read_rows_count', 10)
-        data = self.read_csv_with_delimiter(file, read_rows_count, delimiter=',')
-        return data
 
-    def read_csv_with_delimiter(self, data_stream, read_rows_count: int, delimiter: str = ','):
-        if not delimiter:
-            reader = pd.read_csv(data_stream, sep=None, iterator=True, nrows=read_rows_count)
-            delimiter = reader._engine.data.dialect.delimiter  # pylint: disable=protected-access
-            data_stream.seek(0)
+    def __init__(self, file: BytesIO, **kwargs):
+        self.delimiter = kwargs.get('delimiter') or self._get_delimiter(file)
         try:
-            df = pd.read_csv(data_stream, sep=delimiter)
+            self.data_df = pd.read_csv(file, sep=self.delimiter)
         except ParserError as exc:
             raise DelimiterError() from exc
-        return df
 
-    def get_preview_data(self, df: pd.DataFrame):
-        var_names = list(df.columns)
-        obs_preview = df.head(min(10, df.shape[0]))
-        return var_names, obs_preview
+    def _get_delimiter(self, file: BytesIO):
+        reader = pd.read_csv(file, sep=None, iterator=True, nrows=10)
+        delimiter = reader._engine.data.dialect.delimiter  # pylint: disable=protected-access
+        file.seek(0)
+        return delimiter
 
-    def model_file_validation(self, df: pd.DataFrame, model_metadata_file: BytesIO, need_target: bool = True):
+    def get_obs(self):
```

Comment:
What is obs? Maybe we should use the whole words instead of abbreviation.

Reply:
Changed to observations